lib/tlscontext.h

```diff
@@ -44,6 +44,16 @@ typedef enum
   TVM_REQUIRED=0x0020,
 } TLSVerifyMode;
 
+typedef enum
+{
+  TSO_NONE,
+  TSO_NOSSLv2=0x0001,
+  TSO_NOSSLv3=0x0002,
+  TSO_NOTLSv1=0x0004,
+  TSO_NOTLSv11=0x0008,
+  TSO_NOTLSv12=0x0010,
+} TLSSslOptions;
+
```

Comment:
well, const int's is a lot more typing and I would never want to make
actual variables from these. I dunno if the compiler gets rid off those or
not....

I did check, it doesn't optimize those away:

#include <stdio.h>

const int foobar = 1;

int main()
{
  printf("%d", foobar);
}

bazsi@bzorp:~$ gcc -g a.c -S -o a.i

        .globl  foobar
        .section        .rodata
        .align 4
        .type   foobar, @object
        .size   foobar, 4
foobar:
        .long   1

So it is part of .rodata, though the call-site itself uses the literal
value:

        movl    $1, %eax
        movl    %eax, %esi
        movl    $.LC0, %edi
        movl    $0, %eax
        call    printf




-- 
Bazsi

On Wed, May 20, 2015 at 8:24 PM, Budai Laszlo <notifications@github.com>
wrote:

> In lib/tlscontext.h
> <https://github.com/balabit/syslog-ng/pull/457#discussion_r30733066>:
>
> > @@ -44,6 +44,16 @@ typedef enum
> >    TVM_REQUIRED=0x0020,
> >  } TLSVerifyMode;
> >
> > +typedef enum
> > +{
> > +  TSO_NONE,
> > +  TSO_NOSSLv2=0x0001,
> > +  TSO_NOSSLv3=0x0002,
> > +  TSO_NOTLSv1=0x0004,
> > +  TSO_NOTLSv11=0x0008,
> > +  TSO_NOTLSv12=0x0010,
> > +} TLSSslOptions;
> > +
>
> And what if we have a typedef for TlsSslOption and static const
> TlsSslOption delarations for the values? Or we just change the return
> type of tls_lookup_options to int...
>
> I think the problem with enum is that when we return from
> tls_lookup_options we say that type of the value is TLSSslOptions but
> it's not: it`s a combination of the enumerated values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/457/files#r30733066>.
>